shablona/data.py

```diff
@@ -0,0 +1,144 @@
+"""shablona.data: download and read data."""
+import os
+import sys
+import contextlib
+import os.path as op
+from hashlib import md5
+from shutil import copyfileobj
+
+if sys.version_info[0] < 3:
+    from urllib2 import urlopen, urljoin
+else:
+    from urllib.request import urlopen, urljoin
+
+# Set a user-writeable file-system location to put files:
+SHABLONA_HOME = op.join(os.path.expanduser('~'), '.shablona')
```

Comment:
Thanks for taking a look, and thanks for the comment. What is an XDG compliant directory? 

From a brief look around, it seems that these are often directories like `/usr/local/`, which are not always user-writeable. This needs to be user-writeable! But if I got it all wrong, please educate me. Better yet, a proposal for what to replace this with would be much appreciated.